File: backend/app/schemas/appointment.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field
from app.models.appointment import AppointmentStatus

class AppointmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    patient_id: str
    patient_name: str
    patient_email: str
    provider_id: str
    provider_name: str
    appointment_type: str
    reason: Optional[str] = None
    requested_start: datetime
    scheduled_start: Optional[datetime] = None
    scheduled_end: Optional[datetime] = None
    status: AppointmentStatus
    version: int
    created_at: datetime
    updated_at: datetime

class AppointmentCreateRequest(BaseModel):
    patient_id: str
    patient_name: str
    patient_email: str
    provider_id: str
    provider_name: str
    appointment_type: str = Field(default="General Consultation")
    reason: Optional[str] = None
    requested_start: datetime

class AppointmentConfirmRequest(BaseModel):
    expected_version: int
    scheduled_start: Optional[datetime] = None
    actor_id: Optional[str] = None
    actor_role: Optional[str] = "provider"

class AppointmentRescheduleRequest(BaseModel):
    expected_version: int
    scheduled_start: Optional[datetime] = None
    new_start: Optional[datetime] = None
    actor_id: Optional[str] = None
    actor_role: Optional[str] = "provider"

    @property
    def target_start(self) -> Optional[datetime]:
        return self.scheduled_start or self.new_start

class AppointmentCancelRequest(BaseModel):
    expected_version: int
    actor_id: Optional[str] = None
    actor_role: Optional[str] = "patient"

File: frontend/src/types/appointment.ts
export type AppointmentStatus = 'pending' | 'confirmed' | 'cancelled';

export type Role = 'patient' | 'provider';

export interface Appointment {
  id: number;
  patient_id: string;
  patient_name: string;
  patient_email: string;
  provider_id: string;
  provider_name: string;
  appointment_type: string;
  reason?: string | null;
  requested_start: string;
  scheduled_start?: string | null;
  scheduled_end?: string | null;
  status: AppointmentStatus;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface AuditEvent {
  id: number;
  appointment_id: number;
  actor_id: string;
  actor_role: string;
  action: string;
  before_state?: Record<string, any> | null;
  after_state?: Record<string, any> | null;
  created_at: string;
}

export interface AppointmentCreateRequest {
  patient_id: string;
  patient_name: string;
  patient_email: string;
  provider_id: string;
  provider_name: string;
  appointment_type: string;
  reason?: string;
  requested_start: string;
}

export interface AppointmentConfirmRequest {
  expected_version: number;
  scheduled_start?: string;
  actor_id?: string;
  actor_role?: string;
}

export interface AppointmentRescheduleRequest {
  expected_version: number;
  scheduled_start: string;
  actor_id?: string;
  actor_role?: string;
}

export interface AppointmentCancelRequest {
  expected_version: number;
  actor_id?: string;
  actor_role?: string;
}

export interface ApiErrorDetail {
  code?: string;
  message?: string;
  current_version?: number;
  detail?: string | any;
}
